src/lib/services/instance-stats-service.ts

```diff
@@ -109,6 +115,29 @@ export class InstanceStatsService {
         this.settingStore = settingStore;
         this.clientInstanceStore = clientInstanceStore;
         this.logger = getLogger('services/stats-service.js');
+        process.nextTick(() => {
+            this.refreshStatsSnapshot();
+            this.snapshotRefresher = setInterval(
+                () => this.refreshStatsSnapshot(),
+                minutesToMilliseconds(5),
+            ).unref();
+        });
+    }
+
+    async refreshStatsSnapshot(): Promise<void> {
+        // check if stores are present (required for some tests)
+        if (
+            this.featureToggleStore === undefined ||
+            this.userStore === undefined ||
+            this.clientInstanceStore === undefined
+        ) {
+            return;
+        }
```

Comment:
Without this some tests would fail trying to read from userStore when userStore is undefined. I'm not sure why it happens, but it's also not so bad to validate this

Reply:
I think this is the incorrect approach. I think you instead need to wrap the next line in a try/catch, because the way to start the fetching as an interval a lot of tests will fail due to the database connection being closed just before the refreshStatsSnapshot is called.

Reply:
Oh, that's a good idea! I'll look into it. Thanks!!!

Reply:
The try catch: https://github.com/Unleash/unleash/blob/cb4b765eeea0c47ddaa3ee36f982e2c38d691ef0/src/lib/services/instance-stats-service.ts#L129-L142 doesn't seem to catch `Cannot read properties of undefined` exception types... :thinking: 

I'm not sure how to fix this https://github.com/Unleash/unleash/actions/runs/3849713375/jobs/6559023513#step:5:576

Reply:
https://github.com/Unleash/unleash/pull/2737#discussion_r1062817463